xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/html/HTMLUtils.java

```diff
@@ -289,4 +291,21 @@ public static void stripFirstElementInside(Document document, String parentTagNa
             }
         }
     }
+
+    /**
+     * Escapes HTML special characters in a {@code String} using numerical HTML entities, so that the resulting string
+     * can safely be used as an HTML content text value. For instance, {@code Jim & John} will be escaped and can thus
+     * be put inside a HTML tag, such as the {@code p} tag, as in {@code <p>Jim &amp; John</p>}. Specifically, escapes
+     * &lt; to {@code &lt;}, and &amp; to {@code &amp;}.
+     *
+     * @param content the text to escape, may be {@code null}. The content is converted to a {@link String} using
```

Comment:
Still missing the why? Why did you add a signature that takes an Object instead of a String!

Reply:
I made `content` an `Object` to stay consistent with the other escape methods. But AFAIK we would be fine with a `content` of type `String`. Would it be ok to change the signature of `escapeElementText` to `public static String escapeElementText(String content)`?

Reply:
> I made content an Object to stay consistent with the other escape methods.

But that's not a good-enough reason :) Why did they do that? Imagine they made a mistake, you'd just be continuing the mistake.

Reply:
I think it comes from Velocity.

See:
* Taken from https://github.com/xwiki/xwiki-platform/commit/ed3a5467b08c7069959076ace8e40a2c17bb253c#diff-83c0483287bd455e80eb573959efc65bR47
* And move to https://github.com/xwiki/xwiki-platform/commit/ed3a5467b08c7069959076ace8e40a2c17bb253c#diff-2c6a53f98aefb9159a01c42a82e54255R177

Now need to understand why Velocity tool is using Object. And whether it's needed for a Java API (I don't think so).

Reply:
Not sure what to do. I think i'd be in favor of using String for the new method instead of Object. WDYT?

Reply:
- We control the places where the method is called (and they are all passing string values)
- IMHO, it is a good practice to force the caller to sanitize its input before (instead of dealing with unexpected value in a method).

I'm in favor of using String for the new method.

Reply:
Yep, let's see if @tmortagne and @surli are ok on this too.

Reply:
Otherwise it would mean that everywhere we take a String in APIs we should move to Object and convert to a String. Feel not great and not typed ;)

Reply:
We should probably introduce new signatures for existing methods taking an Object and deprecate existing ones (AFAIK specific types are favored over Object in Java so it should work).

Reply:
I can understand it for Velocity but I don't like this Object parameter for a Java utility method.

Reply:
Each method taking an Object in parameter is now deprecated and is overridden with a method taking a String in parameter